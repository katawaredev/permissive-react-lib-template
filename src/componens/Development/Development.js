/**
 * @param {object} props
 * @param {string} [props.type]
 */
const Development = ({ type = "bundle" }) => (
  <div>This is a development {type}</div>
);

export default Development;
