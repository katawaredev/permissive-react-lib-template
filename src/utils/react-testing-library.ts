import { ReactElement } from "react";
import { RenderOptions, RenderResult, render } from "@testing-library/react";

import AppWrapper from "AppWrapper";

const customRender = (
  ui: ReactElement,
  options?: RenderOptions
): RenderResult => render(ui, { wrapper: AppWrapper, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
