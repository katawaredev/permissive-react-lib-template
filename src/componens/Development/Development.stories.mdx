import { Meta, Story, ArgsTable, Preview } from "@storybook/addon-docs/blocks";
import Development from "./Development";

<Meta title="Development" component={Development} />

# Development

## Preview

export const Template = (args) => <Development {...args} />;

<Preview>
  <Story name="development">{Template.bind({})}</Story>
</Preview>

## Props

<ArgsTable of={Development} />
