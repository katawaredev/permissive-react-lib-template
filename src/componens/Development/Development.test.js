import { render } from "react-testing-library";
import Development from "./Development";

describe("Development component", () => {
  it("matches default snapshot", () => {
    const { asFragment } = render(<Development />, {});
    expect(asFragment()).toMatchSnapshot();
  });
  it("matches custom snapshot", () => {
    const { asFragment } = render(<Development type="custom" />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
