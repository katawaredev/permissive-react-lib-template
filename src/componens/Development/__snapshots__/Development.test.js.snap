// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Development component matches custom snapshot 1`] = `
<DocumentFragment>
  <div>
    This is a development custom
  </div>
</DocumentFragment>
`;

exports[`Development component matches default snapshot 1`] = `
<DocumentFragment>
  <div>
    This is a development bundle
  </div>
</DocumentFragment>
`;
